docs/DeveloperGuide.adoc

```diff
@@ -25,242 +23,382 @@ Refer to the guide <<SettingUp#, here>>.
+* _Alternative 2:_ Update the local storage directly when the command is executing.
+** Pros: No need to implement the internal list.
+** Cons: Will access local memory more frequently. May have performance issues in terms of memory usage.
 
-The `redo` command does the opposite -- it calls `Model#redoAddressBook()`, which shifts the `currentStatePointer` once to the right, pointing to the previously undone state, and restores the address book to that state.
+===== Design considerations: Data structure to support the question commands
+* _Alternative 1 (current choice):_ Use a `UniqueQuestionList` to store questions in `AppData`.
+** Pros: Cater to the question model specifically. Question list operations are encapsulated in one class.
+** Cons: Logic is duplicated as other models also implement similar list structure.
+* _Alternative 2:_ Use Java list to store the questions.
+** Pros: Do not need to maintain a separate list class.
+** Cons: Violates Single Responsibility Principle and Separation of Concerns as the model needs to maintain various
+list operations.
 
-[NOTE]
-If the `currentStatePointer` is at index `addressBookStateList.size() - 1`, pointing to the latest address book state, then there are no undone address book states to restore. The `redo` command uses `Model#canRedoAddressBook()` to check if this is the case. If so, it will return an error to the user rather than attempting to perform the redo.
+=== Quiz mode for revision
+==== Implementation
 
-Step 5. The user then decides to execute the command `list`. Commands that do not modify the address book, such as `list`, will usually not call `Model#commitAddressBook()`, `Model#undoAddressBook()` or `Model#redoAddressBook()`. Thus, the `addressBookStateList` remains unchanged.
+The quiz mode feature is facilitated by `Model`.
+The quiz-related commands extend `Command` with specific question object.
+The commands update the `Model` which is implemented by `ModelManager`.
+This in turn updates `AppData` which stores filtered specific questions internally as `QuizQuestionList`.
+Local data will be updated in the end by `LogicManager`.
+The commands include:
 
-image::UndoRedoState4.png[]
+* `QuizModeCommand` -- Enters the quiz mode with questions selected by user.
+* `QuizCheckAnswer` -- Checks the correctness of answer entered by user.
+* `QuizShowAnswerCommand` -- Shows the answer for current question.
+* `QuitQuizModeCommand` -- Quits the quiz mode.
 
-Step 6. The user executes `clear`, which calls `Model#commitAddressBook()`. Since the `currentStatePointer` is not pointing at the end of the `addressBookStateList`, all address book states after the `currentStatePointer` will be purged. We designed it this way because it no longer makes sense to redo the `add n/David ...` command. This is the behavior that most modern desktop applications follow.
+These operations are exposed in the `Model` interface as `Model#getQuizQuestions(int numOfQuestions, Subject subject
```

Comment:
maybe it would be better to present the methods in a list to make it less cluttered and easier to refer to the previous list?

Reply:
The list is short enough.